requests.go

```diff
@@ -92,16 +92,8 @@ func (c *Client) req(method, path string, body io.Reader, intercept func(*http.R
 }
 
 func (c *Client) mkcol(path string) int {
-	rs, err := c.req("MKCOL", path, nil, nil)
-	if err != nil {
-		return 400
-	}
+	rs, _ := c.req("MKCOL", path, nil, nil)
```

Comment:
Errors returned by the `Client#req` - which also can be others than response errors - should actually not be just ignored.

This could actually raise a nil pointer exception when `rs` is `nil` because of an error early-returned before. And when not that, it could still cause unexpected results.

In my opinion, the best solution would be to bubble up the `error` as additional return value in all of the request functions.

Reply:
You are right about the nil pointers, I have added checks to catch cases where there is no response and return 400 in these cases.

> Errors returned by the Client#req - which also can be others than response errors - should actually not be just ignored.

Agree, but errors are pretty much already ignored.

Reply:
Yeah, error handling is a mess in this package. When I find some time, I'll might look into that. Looks fine to me for now though.